src/Credentials/UserRefreshCredentials.php

```diff
@@ -80,16 +85,22 @@ public function __construct(
             'scope' => $scope,
             'tokenCredentialUri' => self::TOKEN_CREDENTIAL_URI,
         ]);
-        trigger_error(
-            'Your application has authenticated using end user credentials '
-            . 'from Gooogle Cloud SDK. We recommend that most server '
-            . 'applications use service accounts instead. If your application '
-            . 'continues to use end user credentials from Cloud SDK, you might '
-            . 'receive a "quota exceeded" or "API not enabled" error. For '
-            . 'more information about service accounts, see '
-            . 'https://cloud.google.com/docs/authentication/.',
-            E_USER_WARNING
-        );
+        if ($jsonKey['client_id'] === self::CLOUD_SDK_CLIENT_ID
+            && getenv(self::SUPPRESS_CLOUD_SDK_CREDS_WARNING_ENV) !== 'true') {
+            trigger_error(
+                'Your application has authenticated using end user credentials '
```

Comment:
Just a small nit, is it possible to use the same indentation style as the original revision?